packages/core/src/modules/question-answer/services/QuestionAnswerService.ts

```diff
@@ -0,0 +1,250 @@
+      throw new AriesFrameworkError(`Connection with verkey ${connection.verkey} has no recipient keys.`)
+    }
+
+    const questionAnswerRecord = await this.createRecord({
+      questionText: questionMessage.questionText,
+      questionDetail: questionMessage.questionDetail,
+      connectionId: connection?.id,
+      threadId: questionMessage.id,
+      role: QuestionAnswerRole.Responder,
+      signatureRequired: false,
+      state: QuestionAnswerState.QuestionReceived,
+      validResponses: questionMessage.validResponses,
+    })
+
+    await this.questionAnswerRepository.save(questionAnswerRecord)
+
+    this.eventEmitter.emit<QuestionAnswerStateChangedEvent>({
+      type: QuestionAnswerEventTypes.QuestionAnswerStateChanged,
+      payload: { previousState: null, questionAnswerRecord },
+    })
+
+    return questionAnswerRecord
+  }
+
+  /**
+   * create answer message, check that response is valid
+   *
+   * @param questionAnswerRecord record containing question and valid responses
+   * @param response response used in answer message
+   * @returns answer message and QuestionAnswer record
+   */
+  public async createAnswer(questionAnswerRecord: QuestionAnswerRecord, response: string) {
+    const answerMessage = new AnswerMessage({ response: response, threadId: questionAnswerRecord.threadId })
+
+    questionAnswerRecord.response = response
+
+    if (questionAnswerRecord.validResponses.some((e) => e.text === response)) {
+      await this.updateState(questionAnswerRecord, QuestionAnswerState.AnswerSent)
+    } else {
+      this.logger.error(`Response does not match valid responses`)
+    }
+    return { answerMessage, questionAnswerRecord }
+  }
+
+  /**
+   * receive answer as questioner
+   *
+   * @param messageContext the message context containing an answer message message
+   * @returns QuestionAnswer record
+   */
+  public async receiveAnswer(messageContext: InboundMessageContext<AnswerMessage>): Promise<QuestionAnswerRecord> {
+    const { message: answerMessage, connection } = messageContext
+
+    this.logger.debug(`Receiving answer message with id ${answerMessage.id}`)
+
+    if (!connection) {
+      throw new AriesFrameworkError(`Connection for verkey ${messageContext.recipientVerkey} not found!`)
+    }
+
+    if (!connection.theirKey) {
+      throw new AriesFrameworkError(`Connection with verkey ${connection.verkey} has no recipient keys.`)
+    }
+
+    try {
+      const questionAnswerRecord: QuestionAnswerRecord = await this.getByThreadAndConnectionId(
+        answerMessage.threadId,
+        connection?.id
+      )
+
+      await this.updateState(questionAnswerRecord, QuestionAnswerState.AnswerReceived)
+
+      return questionAnswerRecord
+    } catch (error) {
+      this.logger.error(
+        `Unable to get question answer record by threadId "${answerMessage.threadId}" and connection id "${connection?.id}" and unable to update state`,
+        { error }
+      )
+      throw error
```

Comment:
Why the try/catch here? I don't see it in other places. Maybe the base errors are sufficient here?